app/models/script.rb

```diff
@@ -0,0 +1,24 @@
+class Script < ApplicationRecord
+
+  has_many :events
+
+  before_validation :add_or_update_checksum
+
+  validates_presence_of :text
+  validate :unique_text
+
+  private
+
+  def add_or_update_checksum
+    return nil unless self.text # avoid triggering error on Script.new.valid?
+    checksum = Digest::SHA1.hexdigest(self.text)
```

Comment:
Because of recent `SHA1` collisions, we should use `Digest::SHA2`. It's an unlikely attack, but on the other hand it's a 1 character fix.